src/XMakeTasks/FileIO/WriteLinesToFile.cs

```diff
@@ -83,12 +85,12 @@ public override bool Execute()
                     }
                 }
 
-                Encoding encode = null;
+                Encoding encoding = s_defaultEncoding;
                 if (_encoding != null)
```

Comment:
Hmm ... then my follow up suggestion would be to rename `_encoding` or `encoding`.  Makes it easy to mis-read this code 😄